codice/ddf
# DDF-2478 Add service to convert country codes from FIPS 10-4 into ISO 3166-1 alpha 3 and vice versa

#### What does this PR do?

-Adds an OSGi service to convert country codes from FIPS 10-4 into ISO 3166-1 alpha and vice versa.
-LocalCountryCodeConverter.java: loads in a static file (fipsToIso.properties) containing the country code mappings, builds MultiMaps from the mappings to perform the conversions.

-Removes the ISOCountryFormatConverter.java.

This DDF API is designed to be expanded to support not only converting country codes using a local static file, but also converting country codes via external services when available in the future.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?

@dcruver 
@ryeats 
@jlcsmith 
@harrison-tarr 
@peterhuffer 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@lessarderic 
@jaymcnallie
@pklinef
#### How should this be tested?

Run a full build. Make sure integration and unit tests pass. Verify the service install correctly.

Verify the country mappings from this link are the same as the mappings from the country codes mapping file from this PR. I did not include the countries that did not have a FIPS 10-4 to ISO 3166-1 alpha 3 mapping. http://www.opengeocode.org/download/fips2iso.txt

Also, run the `migration:export` admin command and make sure the `fipsToIso3.properties` file gets exported to the `$DDF_HOME/etc/exported/etc` directory.
#### Any background context you want to provide?

There needs to be a way to normalize country codes from inputs eg. NITFs, that contain a 2 letter country code (FIPS 10-4), into a common format DDF uses (ISO 3166-1 alpha 3) so that when a user wants to query a metacard by country, he would search by the normalized country code and still retrieve metacards from inputs that have a country code in FIPS 10-4. This service will provide the ability to normalize those country codes.
#### What are the relevant tickets?

DDF-2478
#### Screenshots (if appropriate)
#### Checklist:
